audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/stream/SerDesConfig.java

```diff
@@ -167,20 +157,25 @@ private SerDesConfig() {
                 return MAPPER.readValue(auditRequest, AuditSuccessMessage.class);
             } catch (IOException e) {
                 String failedAuditString = new String(auditRequest, Charset.defaultCharset());
-                String fileName = "Success-" + ZonedDateTime.now(ZoneOffset.UTC)
-                        .format(DateTimeFormatter.ISO_INSTANT).replace(":", "-");
-                File directory = new File(configProperties.getErrorDirectory());
-                File parent = directory.getAbsoluteFile().getParentFile();
-                File timestampedFile = new File(parent, fileName);
-                try (FileWriter fileWriter = new FileWriter(timestampedFile, Charset.defaultCharset(), !timestampedFile.createNewFile())) {
-                    fileWriter.write(failedAuditString);
-                    LOGGER.info("Failed to deserialize a 'success' audit message. Created file {}", timestampedFile);
-                } catch (IOException ex) {
-                    LOGGER.error("Failed to process audit request '{}'", failedAuditString, ex);
-                }
+                createFile("Success-", failedAuditString, configProperties);
                 SerDesHealthIndicator.addSerDesExceptions(e);
                 throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + failedAuditString, e);
             }
         };
     }
+
+    private static void createFile(final String name, final String failedAuditString, final AuditServiceConfigProperties configProperties) {
```

Comment:
Perhaps change `name` to `prefix`?

Reply:
Yep, my brain wasn't working when coming up with this